### ISSUE TITLE
Update rancher-images.txt generation for airgap

### ISSUE THREAD

**Issue by cbron** [rancher/rancher#27487]
With our movement towards cloud-native-packages / dev-charts, we need to use it to generate rancher-images.txt in a similar manner to how we do today with system-charts. 

The new generator must have a feature that allows certain packages to provide their images via a standard text file, rather than through a search. The package must provide the text file to the generator through a standard naming convention. For example if the generator see's _"images.amd.txt"_ or similar it will use that and skip the search. 

Related: 
* https://github.com/rancher/rancher/issues/19366
* https://github.com/rancher/rancher/issues/22865 
* https://github.com/rancher/rancher/issues/27403

**Comment by cbron** [comment]
Looks like K3s images are missing

**Comment by jiaqiluo** [comment]
This is validated in rancher: `v2.5.0-alpha3`, docker install in airgap environment

k3s images are bundled in the rancher image.

**Comment by cbron** [comment]
Not all the k3s images are bundled with the docker repo. I think we need those to upgrade downstream clusters on a new release, is that right @davidnuzik ? 
That being said I'm not sure if this ticket covers that side of the house or not. Might create separate issue for that.

**Comment by davidnuzik** [comment]
I'm not sure @cbron - my understanding was all I need to do is get k3s in KDM like as per https://github.com/rancher/kontainer-driver-metadata/pull/356 and the k3s images should show up. We are just waiting on approval from @kinarashah to get that in. I've requested re-review as it may not have been clear in the PR we are waiting.
cc: @cjellick @galal-hussein

**Comment by cbron** [comment]
My main question is where are we tracking airgap testing for k3s upgrade since its not currently working. @jiaqiluo @sangeethah do we usually test for that ?
